pypsa/optimization/optimize.py

```diff
@@ -354,7 +354,8 @@ def assign_solution(n):
             else:
                 set_from_frame(n, c, attr, df)
         elif attr != "n_mod":
-            n.df(c)[attr + "_opt"].update(df)
+            idx = df.index.intersection(n.df(c).index)
```

Comment:
Everything looks fine, I just don't quite understand this line, but it has more to do with the fact that I haven't used the modularity feature yet.

Reply:
This is fine.